cogs/guild_join.py

```diff
@@ -0,0 +1,80 @@
+import discord
+from discord.ext import commands
+
+from bot_globals import logger, STREAK_ROLES, MILESTONE_ROLES, VERIFIED_ROLE
+from embeds.misc_embeds import error_embed
+from embeds.stats_embeds import stats_embed, account_hidden_embed
+from embeds.users_embeds import account_not_found_embed
+from models.user_model import User
+from utils.middleware import admins_only, ensure_server_document, track_analytics
+
+class GuildJoin(commands.Cog):
+    def __init__(self, client):
+        self.client = client
+
+    # When the bot joins a server, create a new display information for each user in the server.
+    @commands.Cog.listener()
+    async def on_guild_join(self, guild) -> None:
+        print(f'Joined {guild.name} with {guild.member_count} members!')
+
+        # Create a new role
+        await generate_connected_users_role(self, guild)
+        await generate_milestone_users_role(self, guild)
+        await generate_streaks_users_role(self, guild)
+
+        return None
```

Comment:
You don't need to return None, as functions automatically return None if nothing is explicitly returned